src/System.CommandLine.Tests/GetValueByNameParserTests.cs

```diff
@@ -135,14 +135,88 @@ public void When_required_argument_value_is_not_parsed_then_an_exception_is_thro
         [Fact]
         public void When_non_existing_name_is_used_then_exception_is_thrown()
         {
-            ParseResult parseResult = new Command("noSymbols").Parse("");
+            const string nonExistingName = "nonExisting";
+            Command command = new ("noSymbols");
+            ParseResult parseResult = command.Parse("");
 
-            Action getRequired = () => parseResult.GetValue<int>("required");
+            Action getRequired = () => parseResult.GetValue<int>(nonExistingName);
 
             getRequired
                 .Should()
                 .Throw<InvalidOperationException>()
```

Comment:
`ArgumentException` might be more appropriate here, though I could make an argument for `InvalidOperationException` if the name exists on an unparsed branch of the larger grammar, e.g. under a subcommand that wasn't present in the current result.